### ISSUE TITLE
Deferred initialization clarifications

### ISSUE THREAD

**Issue by marcellomaugeri** [AFLplusplus/AFLplusplus#981]
Hey developers,
I'm trying to fuzz a target that forks itself and I'd like to perform the AFL loop only for the target process, then I'm using the deferred initialization like in this structure:
```
__AFL_FUZZ_INIT();
int main(int argc, char** argv){
    if(fork()==0){ 
        //child process to execute one time only, but it's called many times instead
    } else {
       __AFL_INIT();
       unsigned char *buf = __AFL_FUZZ_TESTCASE_BUF;
       while(__AFL_LOOP(10000)){
           //target to fuzz in a loop
       }
    }
}
```
from what I've understood, the fork call should happen only once and then the forkserver should work only inside the loop, however, the fork is being called many times, slowing the entire process.
Where am I wrong? I'm using afl-clang-fast, but maybe I haven't understood how the deferred initialization works...

**Comment by vanhauser-thc** [comment]
why does the target need to fork on its own? the AFL_INIT needs to be before the fork if the fork is necessary in the target.
note though that the forked part of the program needs to terminate itself at once.
it is better to modify the target so that it does no fork on its own.

**Comment by marcellomaugeri** [comment]
Because in the child process I'd like to perform some extra experiments in parallel in order to improve the coverage and I'm also trying a different strategy. For instance I could fork a server (which has its own loop inside) and a client in the same program, execute the loop only for the client but still get coverage data and crashes from both of them

**Comment by vanhauser-thc** [comment]
@Maray97 I dont think this easily works. also if the target crashes this cannot be detected by afl-fuzz (unless your fork main process also goes into a crash on purpose, etc.)

**Comment by domenukk** [comment]
No more activity. Closing for now

**Comment by marcellomaugeri** [comment]
@abertschi how your commit is referenced to this issue?

**Comment by abertschi** [comment]
> @abertschi how your commit is referenced to this issue?

Must be by mistake. Cheers.
